generators/app/gulpfile.js

```diff
@@ -11,72 +11,90 @@
+          // .pipe(gulpif(/\.css$/, cssSlam()))
+          // .pipe(gulpif(/\.html$/, htmlMinifier()))
+          .pipe(gulpif(/\.(png|gif|jpg|svg)$/, imagemin({
+            progressive: true,
+            interlaced: true
+          })))
+          .pipe(polymerProject.rejoinHtml());
 
-// !!! IMPORTANT !!! //
-// Keep the global.config above any of the gulp-tasks that depend on it
-global.config = {
-  polymerJsonPath: path.join(process.cwd(), 'polymer.json'),
-  build: {
-    rootDirectory: 'build',
-    bundledDirectory: 'bundled',
-    unbundledDirectory: 'unbundled',
-    // Accepts either 'bundled', 'unbundled', or 'both'
-    // A bundled version will be vulcanized and sharded. An unbundled version
-    // will not have its files combined (this is for projects using HTTP/2
-    // server push). Using the 'both' option will create two output projects,
-    // one for bundled and one for unbundled
-    bundleType: 'both'
-  },
-  // Path to your service worker, relative to the build root directory
-  serviceWorkerPath: 'service-worker.js',
-  // Service Worker precache options based on
-  // https://github.com/GoogleChrome/sw-precache#options-parameter
-  swPrecacheConfig: {
-    navigateFallback: '/index.html'
-  }
-};
+        // Okay, now lets do the same to your dependencies
+        let dependenciesStream = polymerProject.dependencies()
+          .pipe(polymerProject.splitHtml())
+          // .pipe(gulpif(/\.js$/, uglify()))
+          // .pipe(gulpif(/\.css$/, cssSlam()))
+          // .pipe(gulpif(/\.html$/, htmlMinifier()))
+          .pipe(polymerProject.rejoinHtml());
 
-// Add your own custom gulp tasks to the gulp-tasks directory
-// A few sample tasks are provided for you
-// A task should return either a WriteableStream or a Promise
-const clean = require('./gulp-tasks/clean.js');
-const images = require('./gulp-tasks/images.js');
-const project = require('./gulp-tasks/project.js');
+        // Okay, now lets merge them into a single build stream
+        let buildStream = mergeStream(sourcesStream, dependenciesStream)
+          .once('data', () => {
+            console.log('Analyzing build dependencies...');
+          });
 
-// The source task will split all of your source files into one
-// big ReadableStream. Source files are those in src/** as well as anything
-// added to the sourceGlobs property of polymer.json.
-// Because most HTML Imports contain inline CSS and JS, those inline resources
-// will be split out into temporary files. You can use gulpif to filter files
-// out of the stream and run them through specific tasks. An example is provided
-// which filters all images and runs them through imagemin
-function source() {
-  return project.splitSource()
-    // Add your own build tasks here!
-    .pipe(gulpif('**/*.{png,gif,jpg,svg}', images.minify()))
-    .pipe(project.rejoin()); // Call rejoin when you're finished
-}
+        // If you want bundling, do some bundling! Explain why?
```

Comment:
Haha, I originally added these comments as placeholders for Rob, if we're committing them we should replace them with a brief explanation.

My hope was that this would become a living document that people could read through to understand each step of their build stream. So +1 on any documentation that helps people better understand what's going / why they would/wouldn't want to include/comment out a line.